### PR TITLE
Add portainer.json

### PR DESCRIPTION
**Pull request by alazare619**
Portainer is an advanced docker cluster,swarm, and single node management framework. Its very advanced and extremely lightweight. It integrates perfectly into rockstor i've been using it for a few weeks now without any issue managing the advanced tasks for my rockstor docker node.

### CODE REVIEW

**Comment by FroggyFlox** [comment]
How does the volume creation/management function on top of Rockstor's shares? I've been tempted and a little interested by Portainer a few weeks ago but wasn't sure how to deal with this part... Would it better to create volume from within Portainer or simply bind to host's share previously created from Rockstor? I would tend to favor the latter in order to keep snapshot possibility, etc..., but I was curious about your experience and use with this so far.

**Comment by magicalyak** [comment]
You're likely going to need to add the --privileged also.  Change the opts to:

```
                "opts": [
                    [
                        "-v",
			"/var/run/docker.sock:/var/run/docker.sock"
                    ],
		    [
			"--privileged",
			""
		    ]
                ],
```

**Comment by magicalyak** [comment]
And I'd add a a volumes map also:

```
                "volumes": {
                    "/data": {
                        "description": "Data share for Portainer information.",
                        "label": "Data Storage"
		    }
		},
```

**Comment by alazare619** [comment]
Weird I didnt get any emails regarding this... Anyways so far I haven't had to have anything else out of the normal. as far as volume maps go yes we could add extra volumes but all the data is actually getting stored in the sock file based on how portainer manages their container...its weird... I haven't needed the privledged yet but I will add that as a CYA

**Comment by alazare619** [comment]
@FroggyFlox  to answer your question I created portainer after I created a bunch of rock-ons already and it managed the existing ones fine. I've created them within portainer as well and they are fine. So either way is fine.

**Comment by alazare619** [comment]
@magicalyak Added the persistent data volume and added it as privileged, I've not had any issues with SELINUX and it yet however without but added it just incase.

@FroggyFlox to answer your question about volumes. I've found it easier to start the dockers by hand via the cli it can use rockstors shares there. Management after the initial creation I do via portainer. I've for some time outgrew the functionality of rockstors management for docker and this definetly supplements it quiet a bit. To clarify I create the shares via rockstor. Then do the initial creation of the docker container via the cli. After that I manage all aspects of the docker via portainer.

File: portainer.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    "description": "Portainer is a lightweight advanced management UI which allows you to easily manage your Docker host or Swarm cluster.",
+    "containers": {
+      "portainer": {
+        "image": "portainer/portainer",
+        "launch_order": 1,
+        "opts": [
+          [
+            "-v",
+            "/var/run/docker.sock:/var/run/docker.sock"
+          ][
```
Comment (doenietzomoeilijk) [inline portainer.json]: There's a comma missing here, making the file invalid JSON.

File: portainer.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            ""
+          ]
+        ],
+        "ports": {
+          "9000": {
+            "description": "Portainer WebUI port. Suggested default: 8090",
+            "host_default": 9000,
+            "label": "WebUI port",
+            "protocol": "tcp",
+            "ui": true,
```
Comment (doenietzomoeilijk) [inline portainer.json]: There's a trailing comma here, making the file invalid JSON.

**Comment by alazare619** [comment]
@schakrava

**Comment by alazare619** [comment]
@phillxnet  can you test this and comit if it all tests well?

**Comment by alazare619** [comment]
closing as this seems to of died.

**Pull request closed without merging**